algorithm.go

```diff
@@ -164,3 +173,20 @@ func (a Algorithm) FromBytes(p []byte) Digest {
 func (a Algorithm) FromString(s string) Digest {
 	return a.FromBytes([]byte(s))
 }
+
+// Validate validates the encoded portion string
+func (a Algorithm) Validate(encoded string) error {
+	r, ok := anchoredEncodedRegexps[a]
+	if !ok {
+		return ErrDigestUnsupported
+	}
+	// Digests much always be hex-encoded, ensuring that their hex portion will
+	// always be size*2
+	if a.Size()*2 != len(encoded) {
```

Comment:
> done, also removed ErrDigestInvalidLength

While I think we may want to keep the `ErrDigestInvalidLength` *constant* [so old consumers still compile][1], I think we want to re-drop this block (unwinding part of your recent 902cd80 → 4ca1301.  Using `.Size() * 2` assumes a hex encoding, and I don't see why callers would need to distinguish between hash portions that are the wrong length (traditionally handled by `ErrDigestInvalidLength`) and hash portions that contain invalid characters (traditionally handled by `ErrDigestInvalidFormat`).  Is there a reason that distinction is useful?

[1]: https://github.com/opencontainers/go-digest/pull/34#discussion_r118374214